.github/workflows/ci-baseline.yml

```diff
@@ -28,75 +28,92 @@ jobs:
   CI:
     name: baseline
     continue-on-error: false
-
     strategy:
-      max-parallel: 10
       fail-fast: false
       matrix:
-        config:
-          # os: Stands for OS on GitHub Runner (Ubuntu / MacOS / Windows)
-          # image: Stands for used Docker Image
-          # mode: Stands for build mode inside Pressio (Debug / Release)
-          - { os: ubuntu-latest, image: ubuntu-20.04-gnu_9-eigen_3.3.7-gtest, mode: Release }
-          - { os: ubuntu-latest, image: ubuntu-20.04-gnu_9-eigen_3.3.7-gtest, mode: Debug }
-
-          - { os: ubuntu-latest, image: ubuntu-20.04-clang_9-eigen_3.3.7-gtest, mode: Release }
-          - { os: ubuntu-latest, image: ubuntu-20.04-clang_9-eigen_3.3.7-gtest, mode: Debug }
-
-          - { os: ubuntu-latest, image: fedora-34-gnu_11-eigen_3.3.7-gtest, mode: Release }
-          - { os: ubuntu-latest, image: fedora-34-gnu_11-eigen_3.3.7-gtest, mode: Debug }
-
-          - { os: ubuntu-latest, image: fedora-34-clang_12-eigen_3.3.7-gtest, mode: Release }
-          - { os: ubuntu-latest, image: fedora-34-clang_12-eigen_3.3.7-gtest, mode: Debug }
-
-          #- { os: ubuntu-latest, image: intel_oneapi-eigen_3.3.7-gtest, mode: Release }
-          #- { os: ubuntu-latest, image: intel_oneapi-eigen_3.3.7-gtest, mode: Debug }
+        image:
+          - ubuntu-20.04-gnu_9-eigen_3.3.7-gtest
+          - ubuntu-20.04-clang_9-eigen_3.3.7-gtest
+          - fedora-34-gnu_11-eigen_3.3.7-gtest
+          - fedora-34-clang_12-eigen_3.3.7-gtest
+          #- intel_oneapi-eigen_3.3.7-gtest
+        build_type:
+          - Release
+          - Debug
+        cmake_extra_args:
+          - ''
+        include:
+          - image: ubuntu-20.04_gnu-11_eigen-3.3.7_gtest
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
+          - image: ubuntu-20.04_clang-12_eigen-3.3.7_gtest
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
+          - image: fedora-34-gnu_11-eigen_3.3.7-gtest
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
+          - image: fedora-34-clang_12-eigen_3.3.7-gtest
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
```

Comment:
why are you putting c++20 here? we need to test for C++14, 17 and 20

Reply:
This is to avoid repetition - we achieve the same coverage without duplicating the workflow file.